Update version and CHANGELOG to 1.2.0 (#159)
heremaps/here-olp-sdk-typescript: CHANGELOG.md

## v1.2.0 (04/02/2020)

**Common**

* Updated the development dependencies.

**olp-sdk-dataservice-read**

* Added the `IndexLayerClient` class that is used to access index layers on OLP. This class implements the `getPartitions` and `getData` methods.

> * Added the `HttpError` class that extending Error class and adding status code to the http errors.

Reviewer: ```suggestion
* Added the `HttpError` class that extends the `Error` class and adds a status code to HTTP errors.
```

> * Improved errors propagations in all public methods. Now the methods rejects the Promises with instance of `HttpError` or `Error` class in case of error.

Reviewer: ```suggestion
* Improved error propagations in all public methods. Now, these methods reject the promises with the `HttpError` instance or `Error` class when errors occur.
```

**olp-sdk-authentication**

> * Added the `HttpError` class that extending Error class and adding status code to the http errors.

Reviewer: ```suggestion
* Added the `HttpError` class that extends the `Error` class and adds a status code to HTTP errors.
```

> * Improved errors propagations in all public methods. Now the methods rejects the Promises with instance of `HttpError` or `Error` class in case of error.

Reviewer: ```suggestion
* Improved error propagations in all public methods. Now, these methods reject the promises with the `HttpError` instance or `Error` class when errors occur.
```

> * **Breaking Change** Improved return type for the `getEarliestVersion` method in `CatalogClient`. Return data type changed to be the same as `getLatestVersion` method.

Reviewer: ```suggestion
* **Breaking Change** Improved the return type of the `getEarliestVersion` method in `CatalogClient`. It is now the same as the return type of the `getLatestVersion` method.
```

> * **Breaking Change** Public methods rejects the Promises with instance of `HttpError` or `Error` class instead of strings in case of error.

Reviewer: ```suggestion
* **Breaking Change** Public methods now reject the promises with the `HttpError` instance or `Error` class instead of strings when errors occur.
```

**olp-sdk-authentication**

> * Added a possibility to read credentials from a file when using Node.js.

Reviewer: ```suggestion
* Added a possibility to read credentials from a file using Node.js.
```
